src/App.tsx

```diff
@@ -1,26 +1,177 @@
+import { TodoList } from './components/ToDoList';
+import { TodoHeader } from './components/ToDoHeader';
+import { TodoFooter } from './components/ToDoFooter';
+import { Error } from './components/Error';
+import { type Todo } from './types/Todo';
+import { createTodo } from '../src/components/api/todos';
+import { FilterButtons } from './types/FilterType';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+  const [tempIds, setTempIds] = useState<number[]>([]);
+  const [newToDoTitle, setNewToDoTitle] = useState('');
+  const [filterButton, setFilterButton] = useState<FilterButtons>(
+    FilterButtons.All,
+  );
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setError('Unable to load todos');
+      });
+  }, []);
+
+  const handleAddingTodos = (event: React.FormEvent<Element>) => {
+    event.preventDefault();
+    if (!newToDoTitle.trim()) {
+      setError('Title should not be empty');
+
+      return;
+    }
+
+    const createdTodo = {
+      id: 0,
+      userId: USER_ID,
+      title: newToDoTitle,
+      completed: false,
+      editted: false,
+    };
+
+    setTemporaryTodo(createdTodo);
+    setTempIds(currentIds => [...currentIds, 0]);
+
+    return createTodo(createdTodo)
+      .then(curTodo => {
+        setTodos([...todos, curTodo]);
+        setNewToDoTitle('');
+      })
+      .catch(() => {
+        setError('Unable to add a todo');
+      })
+      .finally(() => {
+        setTemporaryTodo(null);
+        setTempIds(currentIds => currentIds.filter(todoId => todoId !== 0));
+      });
+  };
+
+  const handleDeletedTodo = (idNumber: number) => {
+    setTempIds(currentIds => [...currentIds, idNumber]);
+
+    return deleteTodo(idNumber)
+      .then(() => {
+        setTodos(updateTodos =>
+          updateTodos.filter(todo => todo.id !== idNumber),
+        );
+      })
+      .catch(() => {
+        setError('Unable to delete a todo');
+      })
+      .finally(() => {
+        setTempIds(currentIds =>
+          currentIds.filter(deletedId => idNumber !== deletedId),
+        );
+      });
+  };
+
+  const deleteOnlyCompleted = () => {
+    const removeDeletedTodos = todos.filter(todo => !todo.completed);
+
+    setTodos(removeDeletedTodos);
+  };
+
+  const filteredTodos = (
+    filtrTodos: Todo[],
+    filterStatus: FilterButtons,
+  ): Todo[] => {
+    const updateTodos = [...filtrTodos];
+
+    if (filterStatus) {
+      switch (filterStatus) {
+        case FilterButtons.Active:
+          return updateTodos.filter(todo => !todo.completed);
+        case FilterButtons.Completed:
+          return updateTodos.filter(todo => todo.completed);
+        default:
+          break;
+      }
+    }
+
+    return updateTodos;
+  };
```

Comment:
To improve your code consider moving the filtering logic to the ToDoList component and updating it only when the todos or filter change. This will separate concerns more effectively and enhance performance